src/rust/ide/view/graph-editor/src/component/node/vcs.rs

```diff
@@ -0,0 +1,202 @@
+            Status::Unchanged => style.get_color(ensogl_theme::graph_editor::node::vcs::unchanged),
+            Status::Added     => style.get_color(ensogl_theme::graph_editor::node::vcs::added),
+            Status::Edited    => style.get_color(ensogl_theme::graph_editor::node::vcs::edited),
+        }
+    }
+}
+
+impl Default for Status {
+    fn default() -> Self {
+        Status::Unchanged
+    }
+}
+
+
+
+// =======================
+// === Indicator Shape ===
+// =======================
+
+/// Shape used in the status indicator. Appears as a colored border surrounding the node.
+mod status_indicator_shape {
+    use super::*;
+
+    const INDICATOR_WIDTH_OUTER : f32 = 15.0;
+    const INDICATOR_WIDTH_INNER : f32 = 10.0;
+
+    ensogl::define_shape_system! {
+        (style:Style,color_rgba:Vector4<f32>) {
+            let width  = Var::<Pixels>::from("input_size.x");
+            let height = Var::<Pixels>::from("input_size.y");
+            let width  = width  - node::PADDING.px() * 2.0;
+            let height = height - node::PADDING.px() * 2.0;
+            let radius = node::RADIUS.px();
+
+            let base = Rect((&width,&height)).corners_radius(&radius);
+            let outer = base.grow(INDICATOR_WIDTH_OUTER.px());
+            let inner = base.grow(INDICATOR_WIDTH_INNER.px());
+
+            (outer-inner).fill(color_rgba).into()
+        }
+    }
+}
+
+
+
+// ==============================
+// === Status Indicator Model ===
+// ==============================
+
+/// Internal data of `StatusIndicator`.
+#[derive(Clone,CloneRef,Debug)]
+struct StatusIndicatorModel {
+    shape : ShapeView<status_indicator_shape::Shape>,
+    root  : display::object::Instance,
+}
+
+impl StatusIndicatorModel {
+    fn new(logger: &Logger, app:&Application) -> Self {
+        let scene  = app.display.scene();
+        let shape = ShapeView::<status_indicator_shape::Shape>::new(&logger,scene);
+        let root = display::object::Instance::new(&logger);
+        root.add_child(&shape);
+        StatusIndicatorModel{shape, root}
+    }
+
+    fn hide(&self) {
+        self.shape.unset_parent();
+    }
+
+    fn show(&self) {
+        self.root.add_child(&self.shape);
+    }
+
+    fn set_visibility(&self, visibility:bool) {
+        if visibility {
+            self.show()
+        } else {
+            self.hide()
+        }
+    }
+}
+
+impl display::Object for StatusIndicatorModel {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.root
+    }
+}
+
+
+
+// =======================
+// === StatusIndicator ===
+// =======================
+
+ensogl::define_endpoints! {
+    Input {
+        set_status     (Option<Status>),
+        set_size       (Vector2),
+        set_visibility (bool),
+    }
+    Output {
+        status (Option<Status>),
+    }
+}
+
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct StatusIndicator {
+        model : Rc<StatusIndicatorModel>,
+    pub frp   : Frp,
```

Comment:
we are not doing such alignment anywhere. Lets not use it + if you think its good way of coding, lets create a PR to the styling docs and lets do discussion there, please.